src/native-stack/views/NativeStackView.tsx

```diff
@@ -129,13 +183,18 @@ export default function NativeStackView({
                 target: key,
               });
             }}>
-            <HeaderConfig {...options} route={route} />
-            <Container
-              style={viewStyles}
-              // @ts-ignore Wrong props passed to View
-              stackPresentation={stackPresentation}>
-              {renderScene()}
-            </Container>
+            <HeaderConfig
+              {...options}
+              route={route}
+              headerShown={isHeaderInPush}
```

Comment:
It looks like `isHeaderInPush` variable is redundant. I think when it's `InModal` it can't be `InPush` at the same time.
```suggestion
              headerShown={!isHeaderInModal}
```

Reply:
But it can also be none of them with `headerShown === false`.